clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -1997,7 +2001,7 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::open_reader(
         RecordType record = **reader;
         if (record_type_has_tid(record, tid))
             break;
-        input.queue.push_back(record);
+        input.queue.emplace_back(record, reader->get_next_trace_pc());
```

Comment:
We already consumed the next record from the reader via operator*, so is this next PC the next-next PC instead of the next PC?